enforcer-rules/src/main/java/org/apache/maven/plugins/enforcer/RequireUpperBoundDeps.java

```diff
@@ -236,7 +237,17 @@ private String getFullArtifactName( DependencyNode node, boolean usePremanaged )
         {
             version = uniqueVersions ? artifact.getVersion() : artifact.getBaseVersion();
         }
-        return artifact.getGroupId() + ":" + artifact.getArtifactId() + ":" + version;
+        String result = artifact.getGroupId() + ":" + artifact.getArtifactId() + ":" + version;
+        if ( artifact.getScope() != null )
```

Comment:
Based on the eclipse output you only want the conflicts to be highlighted, not the entire tree.
I'd advice to add a screenshot of the result to ensure we're working towards the same result.

Reply:
![requireUpper](https://user-images.githubusercontent.com/1631042/60114012-52426300-9749-11e9-80b2-afa14990f0e8.png)

Eclipse show in black font the compile dependencies and in grey color other scopes (test, runtime). I added the font highlight to the compile dependencies to resemble this, but I'm open to suggestions.